cpp/ql/src/semmle/code/cpp/ir/dataflow/DefaultTaintTracking.qll

```diff
@@ -153,6 +159,11 @@ private predicate nodeIsBarrier(DataFlow::Node node) {
   )
 }
 
+private predicate nodeIsBarrierIn(DataFlow::Node node) {
+  // don't use dataflow into taint sources, as this leads to duplicate results.
+  isUserInput(node.asExpr(), _)
```

Comment:
If you want `nodeIsBarrierIn` to be all sources, then `getNodeForSource` is the right predicate to call.

Reply:
Yes, that's a better choice (updated).